dawn/src/dawn/CodeGen/Cuda/MSCodeGen.cpp

```diff
@@ -1059,15 +1055,13 @@ void MSCodeGen::generateCudaKernelCode() {
           std::string iterators = "IJ";
           for(const auto& stage : iterateIIROver<iir::Stage>(*(stencilInstantiation_->getIIR()))) {
             std::string prefix = "stage" + std::to_string(stage->getStageID()) + "Global";
-            int index = 0;
-            for(const auto& interval : stage->getIterationSpace()) {
+            for(auto [index, interval] : enumerate(stage->getIterationSpace())) {
```

Comment:
I am torn if this is more or less readable than the standard `for(int i=0; i< size;++i)`
it is of course fancier and it's nice that c++ offers a way to get indices from range base loops. Is there an an advantage to this or do you think it's just the nicer way of doing things?

Reply:
Johann has encouraged me to use these modern iterators, I personally have no strong preference. I am happy to change to standard C++ loops if you like.